File: Php/fourier_transform.php
<?php
// Fourier Transform in PHP
// This script calculates the Fourier Transform of a sample signal and 
// prints the frequency spectrum.

define('PI', 3.14159265358979323846);
$Fs = 1000;  // Sampling frequency
$L = 1000;   // Length of signal

// Generate a signal composed of two sinusoids
$signal = [];
for ($i = 0; $i < $L; $i++) {
    $t = $i / $Fs;
    $signal[$i] = 0.7 * sin(2 * PI * 50 * $t) + sin(2 * PI * 120 * $t);
}

// Compute the Fourier Transform using a simple DFT
$real = array_fill(0, $L, 0.0);
$imag = array_fill(0, $L, 0.0);
for ($k = 0; $k < $L; $k++) {
    for ($n = 0; $n < $L; $n++) {
        $real[$k] += $signal[$n] * cos(2 * PI * $k * $n / $L);
        $imag[$k] -= $signal[$n] * sin(2 * PI * $k * $n / $L);
    }
}

// Print frequency spectrum (magnitude only)
for ($k = 0; $k < $L / 2; $k++) {
    $magnitude = sqrt($real[$k] * $real[$k] + $imag[$k] * $imag[$k]) / $L;
    $frequency = ($Fs * $k) / $L;
    printf("Frequency: %.1f Hz, Magnitude: %.2f\n", $frequency, $magnitude);
}
?>

File: Fortran/fourier_transform.f90
! Fourier Transform in Fortran
! This program calculates the Fourier Transform of a sample signal and
! prints the frequency spectrum.

program fourier_transform
  implicit none
  integer, parameter :: L = 1000
  real, parameter :: Fs = 1000.0
  real :: signal(L), real_part(L), imag_part(L)
  integer :: i, j
  real :: freq, t, magnitude

  ! Generate a signal composed of two sinusoids
  do i = 1, L
     t = (i - 1) / Fs
     signal(i) = 0.7 * sin(2.0 * 3.14159 * 50.0 * t) + &
                 sin(2.0 * 3.14159 * 120.0 * t)
  end do

  ! Initialize real and imaginary parts for Fourier Transform
  real_part = 0.0
  imag_part = 0.0

  ! Compute the Fourier Transform using a simple DFT
  do i = 1, L
     do j = 1, L
        real_part(i) = real_part(i) + signal(j) * cos(2.0 * 3.14159 * (i-1) * (j-1) / L)
        imag_part(i) = imag_part(i) - signal(j) * sin(2.0 * 3.14159 * (i-1) * (j-1) / L)
     end do
  end do

  ! Print frequency spectrum (magnitude only)
  do i = 1, L/2
     freq = Fs * (i - 1) / L
     magnitude = sqrt(real_part(i)**2 + imag_part(i)**2) / L
     print *, 'Frequency:', freq, 'Hz, Magnitude:', magnitude
  end do

end program fourier_transform
